pkg/certs/certs_test.go

```diff
@@ -0,0 +1,99 @@
+package certs
+
+import (
+	"crypto/x509/pkix"
+	"io/ioutil"
+	"testing"
+	"time"
+)
+
+func TestValidateCertificates(t *testing.T) {
+	c, err := NewCACertificate(pkix.Name{CommonName: "test"}, int64(1), time.Now)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(c.Certificates) != 1 {
+		t.Fatalf("expected 1 certificate in the chain, but got %d", len(c.Certificates))
+	}
+
+	validCerts, err := FilterExpiredCerts(c.Certificates...)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(validCerts) != 1 {
```

Comment:
you should have a negative test too. Shouldn't be too hard to get an expired cert.